modules/renter/download.go

```diff
@@ -191,33 +191,41 @@ func (r *Renter) Download(path, destination string) error {
 
 	// Look up the most recent contract for each host.
 	// NOTE: this assumes that only one contract is made with each host.
-	contracts := make(map[*fileContract]modules.RenterContract)
+	var contractPieces []struct {
+		contract modules.RenterContract
+		pieces   []pieceData
+	}
 	file.mu.RLock()
 	for _, fc := range file.contracts {
```

Comment:
it would still be ugly:
```go
for id := range file.contracts {
    fc := file.contracts[id]
    contracts[&fc] = c
}
```
and a newcomer might not understand the rationale.
Mostly I didn't like the old map anyway; using a `*fileContract` as a map key was basically just a way to avoid defining a new type.